test/src/main/java/org/axonframework/test/fixture/AxonTestPhase.java

```diff
@@ -0,0 +1,554 @@
+        }
+
+        /**
+         * Configures a single command with the given {@code payload} and {@code metaData} as part of the "given" state.
+         * This command will be dispatched to corresponding command handlers.
+         *
+         * @param payload  The payload of the command to dispatch.
+         * @param metaData The metadata to attach to the command.
+         * @return the current Given instance, for fluent interfacing.
+         */
+        default Given command(Object payload, Map<String, ?> metaData) {
+            return command(payload, MetaData.from(metaData));
+        }
+
+        /**
+         * Configures a single command with the given {@code payload} and {@code metaData} as part of the "given" state.
+         * This command will be dispatched to corresponding command handlers.
+         *
+         * @param payload  The payload of the command to dispatch.
+         * @param metaData The metadata to attach to the command.
+         * @return the current Given instance, for fluent interfacing.
+         */
+        Given command(Object payload, MetaData metaData);
+
+        /**
+         * Configures the given {@code messages} as commands in the "given" state. These commands will be dispatched in
+         * the order they are provided in the same Unit of Work.
+         *
+         * @param messages The command messages to dispatch.
+         * @return the current Given instance, for fluent interfacing.
+         */
+        Given commands(CommandMessage<?>... messages);
+
+        /**
+         * Transitions to the "when" phase to execute the test action. This method completes the "given" phase,
+         * committing any Unit of Work that was started during this phase.
+         *
+         * @return a {@link When} instance that allows executing the test.
+         */
+        When when();
+
+        /**
+         * Transitions to the "when" phase with a consumer to define the test execution. This method completes the
+         * "given" phase, committing any Unit of Work that was started during this phase.
+         *
+         * @param onWhen Consumer to configure the "when" phase.
+         * @return a {@link When} instance, configured according to the provided consumer.
+         */
+        default When when(Consumer<AxonTestPhase.When> onWhen) {
+            var when = when();
+            onWhen.accept(when);
+            return when;
+        }
+    }
+
+    /**
+     * Interface describing the operations available in the "when" phase of the test fixture execution. This phase is
+     * used to execute the actual action being tested, typically a command.
+     * <p>
+     * Each operation in the phase (such as dispatching a command) is executed in its own separate Unit of Work which is
+     * committed immediately after execution. This allows for building up the initial state incrementally with each
+     * operation being processed independently.
+     */
+    interface When {
```

Comment:
Should we limit When phase to just one invocation? 
If yes, Then phase assertions can be based on what we have triggered in When phase. 
Eg. when you have command in when phase - then you can assert command results, if you have event -> you cannot.

Reply:
> Should we limit When phase to just one invocation?
I do think so, yes.

> If yes, Then phase assertions can be based on what we have triggered in When phase.
> Eg. when you have command in when phase - then you can assert command results, if you have event -> you cannot.
That's a nice idea actually!

Reply:
Done